### PR TITLE
Milestone 2: Setup redux store

### PR DESCRIPTION
**Pull request by prg-04**
### I have done the following to reach this milestone:

- [ ] Added Redux Toolkit
- [ ] Created a directory that contains all my Redux logic (/src/redux)
- [ ] Configured a Redux store (/src/redux/store.js)
- [ ] Defines a slice of state for books that:
  -  Includes an array of books (initial state: empty array)
  -  Includes a reducer that adds a book
  -  Includes a reducer that removes a book
- [ ] Defined a slice of state for categories that:
  -  Includes an array of categories (initial state: empty array)
  - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/components/Form.jsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
 import React, { useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import { v4 as uuidv4 } from 'uuid';
-import PropTypes from 'prop-types';
+// import PropTypes from 'prop-types';
```
Comment (rustam-tolipov) [inline src/components/Form.jsx]: Your code is working as expected! 

- [ ] But we are not allowed to use unused commented code in this project! Please remove it.

File: src/components/Form.jsx [hunk 2]
```diff
@@ -33,17 +36,31 @@ const Form = ({ setBooks }) => {
...
+        />
         <button type="submit">Add book</button>
       </form>
     </div>
   );
 };
 
 Form.propTypes = {
-  setBooks: PropTypes.func.isRequired,
+  // setBooks: PropTypes.func.isRequired,
```
Comment (rustam-tolipov) [inline src/components/Form.jsx]: - [ ] Kindly remove unused commented code from your project :)

File: src/components/Book.jsx [hunk 3]
```diff
@@ -1,13 +1,23 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
 import { CircularProgressbar } from 'react-circular-progressbar';
+import { removeBook } from '../redux/books/booksSlice';
```
Comment (rustam-tolipov) [inline src/components/Book.jsx:5]: - [ ] Also, in your readme this issue link is broken. Kindly fix it and make sure it directs to correct link.


[Link to highlight](https://github.com/prg-04/bookstore_cms./tree/redux#:~:text=Feel%20free%20to%20check%20the%20issues%20page.)

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @prg-04,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great readme_
_Project is working as expected_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shisui6 — APPROVED** [review]
Hi @prg-04, 

# Approved 🪄
Your project is complete! There is nothing else to say other than... it's time to merge it 💎
Congratulations! 🎉 

## Highlights 🎯
- Descriptive PR title and summary ✅
- Follows Github flow ✅
- Good commit messages ✅

## Optional suggestions 🪶

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**